specs/LocaleRegion.md

```diff
@@ -0,0 +1,111 @@
+2-letter code from [ISO 639](https://www.iso.org/iso-639-language-codes.html) and `country` is the 
+2-letter code from [ISO 3166](https://www.iso.org/standard/72482.html).
+
+# Examples
+```cpp
+auto webViewEnvironment10 = m_webViewEnvironment.try_query<ICoreWebView2Environment10>();
+wil::com_ptr<ICoreWebView2ControllerOptions> options;
+HRESULT hr = webViewEnvironment10->CreateCoreWebView2ControllerOptions(&options);
+options->put_LocaleRegion(m_webviewOption.localeRegion.c_str());
+```
+
+```c#
+CoreWebView2Environment environment;
+CoreWebView2ControllerOptions CreateCoreWebView2ControllerOptions(CoreWebView2Environment environment)
+{
+    CoreWebView2ControllerOptions ControllerOptions = null;
+    if (LocaleRegion != null)
+    {
+        ControllerOptions = environment.CreateCoreWebView2ControllerOptions();
+        ControllerOptions.LocaleRegion = LocaleRegion;
+    }
+    return ControllerOptions;    
+}
+```
+
+# API Details
+```cpp
+[uuid(0c9a374f-20c3-4e3c-a640-67b78a7e0a48), object, pointer_default(unique)]
+interface ICoreWebView2StagingControllerOptions : IUnknown {
+  /// The default region for the WebView2.  It applies to JavaScript API
+  /// Intl.DateTimeFormat() which affects string formatting like
+  /// in the time/date formats. The intended locale value is in the format of
+  /// `language-country` or `language_country` where `language` is the 2-letter code from [ISO
+  /// 639](https://www.iso.org/iso-639-language-codes.html) and `country` is the
+  /// 2-letter code from [ISO 3166](https://www.iso.org/standard/72482.html).
+  ///
+  /// This property sets the region for a CoreWebView2Environment during its creation. 
+  /// Creating a new CoreWebView2Environment object that connects to an already running 
+  /// browser process cannot change the region previously set by an earlier CoreWebView2Environment.  
+  /// The CoreWebView2Environment and all associated webview2 objects will need to closed.
+  ///
+  /// The default value for LocaleRegion will be depend on the WebView2 language
+  /// and OS region. If the language portions of the WebView2 language and OS Region
+  /// match, then it will use the OS region. Otherwise, it will use the WebView2
+  /// language.
+  /// | **OS Region** | **WebView2 Language** | **Default WebView2 LocaleRegion** |
+  /// |-----------|-------------------|-------------------------------|
+  /// | en-GB     | en-US             | en-GB                         |
+  /// | es-MX     | en-US             | en-US                         |
+  /// | en-US     | en-GB             | en-US                         |
+  /// The default value can be reset using the empty string.
+  ///
+  /// Use OS specific APIs to determine the OS region to use with this property
+  /// if you want to match the OS. For example:
+  ///
+  /// Win32 C++:
+  /// ```cpp
+  ///   int LanguageCodeBufferSize =
+  ///       ::GetLocaleInfoEx(LOCALE_NAME_USER_DEFAULT, LOCALE_SNAME, nullptr, 0);

```

Comment:
Ideally you would use the UPLL (User Preferred Language List) APIs.

Reply:
Was looking through UPLL documentation and didn't see anything to grab the OS region. This example of GetLocaleInfoEx() is what the Edge browser does when matching on OS region as part of their setting.

Reply:
UPLL lets you get the user's preferred set of languages, for example "I prefer English, but I also speak German and know a little bit of Japanese." An app that is localized only for (say) Korean or Japanese content would not want to pick English (it's meaningless) but would want to pick Japanese.

Reply:
A majority of the customers' asks are regarding using the regional format found under Settings->Region in the OS, which could be completely different from the display language. For example, the language could be in English ("en-US") but they want their date and time formatting to be "13/06/2022 18:01:00" ("es-MX" format). This only affects the formatting of certain strings defined by the Intl namespace in the JS object.

Reply:
The "right" answer here is either to use GetUserDefaultLocaleName() (the win32 answer), or the LPLL.  Windows.Globalization.ApplicationLanguages.Languages is probably the right answer.

Windows.System.UserProfile.GlobalizationPreferences.Languages also has the user's preferred list.  And Windows.Globalization.ApplicationLanguages.PrimaryLanguageOverride has what's been selected for that app.

I really don't like the latter as it's inconsistently used.  In Win8 we meant to have a system setting for app properties that would allow users to pick their preferred language for an app, but the UX was cut.  Apps really shouldn't be using that unless we get the system thing hooked up.  We also totally mess up resolution for "system" apps so our LPLL behavior isn't always predictable.  You have to know if it's a system or normal app.

Anyhoo, pick one and use that.  Don't try to do anything further.

Reply:
Just tried using GetUserDefaultLocaleName and this also works as intended. I will update this document to reference that API instead of GetLocaleInfoEx().